base/file.jl

```diff
@@ -341,37 +354,47 @@ Obtain the path of a temporary directory (possibly shared with other processes).
 tempdir()
 
 """
-    tempname()
+    tempname(; prefix="$temp_prefix")
+
+Generate a unique temporary file path. A `prefix` can be provided in addition, which
+otherwise defaults to "$temp_prefix".
 
-Generate a unique temporary file path.
+*Note: The `prefix` may be truncated based on the limitations of the OS.*
```

Comment:
Given what you noted about POSIX, and since Windows also truncates, it would make sense to say something like "will generally be truncated to three or five characters". Also better mention this in the sentence about `prefix` as it's not an interesting note if you don't use it (as I said before, no need to repeat the default).